deploy/build-deploy-common.sh

```diff
@@ -160,9 +160,9 @@ build_image() {
 
     if [ -n "$BUILD_ARGS" ]; then
         BUILD_ARGS_CMD=$(_get_build_args)
-        container_engine_cmd build --pull -f "$DOCKERFILE" $BUILD_ARGS_CMD -t "${IMAGE_NAME}:${IMAGE_TAG}" .
+        container_engine_cmd build --target=compliance-backend --pull -f "$DOCKERFILE" $BUILD_ARGS_CMD -t "${IMAGE_NAME}:${IMAGE_TAG}" .
```

Comment:
please move this logic to the `BUILD_ARGS_CMD` variable. the `build-deploy-common.sh` is supposed to be a library of shared use and not meant to be updated regularly. 

the `BUILD_ARGS_CMD` variable is meant to be used here: https://github.com/RedHatInsights/compliance-backend/blob/442b82959741d82c9fdda58ab4b9d7e8bf70f296/build_deploy.sh#L10

Reply:
:thinking: I can do that, but then we would need to run the command twice probably, for both targets :thinking:

Reply:
Sorry, I didn't realize the `BUILD_ARGS_CMD` wouldn't actually work as it is not a Dockerfile build argument, but a `docker/podman` build argument. 

However your approach may not have the effect you want, see https://docs.docker.com/engine/reference/commandline/build/#specifying-target-build-stage---target

Just reorganize the commands order so the `--target` gets to be the last one used please. 

Also, while you're at it, try to fix the fact that you have to forcibly specify now a `target` for the build, when we actually didn't need to before. 

I'd suggest also to refactor the `if` clause that checks if there's build arguments , remove that check entirely as the result should be the same, as the `BUILD_ARGS_CMD` variable will be empty if not used, and use it to instead provide the `--target` option if the variable is provided. 

As a final nitpick - can you rename the variable to something more close to what it is, like `TARGET_BUILD_STAGE` or similar ? 

Thank you !

Reply:
so with the change as it is now,  a `TARGET_BUILD_STAGE` variable is mandatory, which wasn't the case before. 

I think we could avoid this by adding an "if-else" clause to check if it's defined , and if it's not, don't pass the `--target` parameter at all. 

And to not to overly complicate the current logic, I think the if clause for build args is irrelevant, and as an empty `BUILD_ARGS`variable will not break the build command.

so I think setting the code like 

```
    local BUILD_ARGS_CMD=''

    if [ -n "$BUILD_ARGS" ]; then
        BUILD_ARGS_CMD=$(_get_build_args)
    fi

    if [ -n "$TARGET_BUILD_STAGE" ]; then
        container_engine_cmd build --pull -f "$DOCKERFILE" $BUILD_ARGS_CMD -t "${IMAGE_NAME}:${IMAGE_TAG}" --target=${TARGET_BUILD_STAGE} .

    else
        container_engine_cmd build --pull -f "$DOCKERFILE" $BUILD_ARGS_CMD -t "${IMAGE_NAME}:${IMAGE_TAG}" .
    fi
```

should work without creating a new dependency on `--target`.